File: src/model/record.py
class Record:
    def __init__(
            self,
            id=0,
            uid=0,
            path="",
            request_time=0,
            finished_time=0,
            result=-1) -> None:
        self.id = id
        self.uid = uid
        self.path = path
        self.request_time = request_time
        self.finished_time = finished_time
        self.result = result

    def to_json(self):
        return {
            "id": self.id,
            "uid": self.uid,
            "path": self.path,
            "request_time": self.request_time,
            "finished_time": self.finished_time,
            "result": self.result
        }

File: ui/app/src/main/java/com/luozi/fireeyewatcher/model/Record.java
package com.luozi.fireeyewatcher.model;

import com.luozi.fireeyewatcher.utils.DateUtil;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Date;

public class Record {
    public int id;
    public int uid;
    public String requestTime;
    public String finishedTime;
    public int result;

    public Record(int id, int uid, String requestTime, String finishTime, int result) {
        this.id = id;
        this.uid = uid;
        this.requestTime = requestTime;
        this.finishedTime = finishTime;
        this.result = result;
    }

    public static Record parseFromJson(JSONObject json) throws JSONException {
        return new Record(
                json.getInt("id"),
                json.getInt("uid"),
                DateUtil.convert(json.getInt("request_time")),
                json.getInt("finished_time") == 0 ? "" : DateUtil.convert(json.getInt("finished_time")),
                json.getInt("result")
        );
    }
}
